addon/modifiers/did-intersect.js

```diff
@@ -1,44 +1,56 @@
 import Modifier from 'ember-modifier';
+import { inject as service } from '@ember/service';
+import { assert } from '@ember/debug';
 
 export default class DidIntersectModifier extends Modifier {
-  // Public API
-  handler = null;
-  options = {};
+  @service('ember-scroll-modifiers@observer-manager') observerManager;
 
-  // Private API
-  observer = null;
+  _isObservable = true;
 
-  observe() {
-    if ('IntersectionObserver' in window) {
-      this.observer = new IntersectionObserver((entries, observer) => {
-        this.handler(entries[0], observer);
-      }, this.options);
+  constructor() {
+    super(...arguments);
+    this._isObservable = 'IntersectionObserver' in window;
+  }
 
-      this.observer.observe(this.element);
+  observe(options) {
+    if (!this._isObservable) {
+      return;
     }
+
+    this.observerManager.observe(this.element, options);
   }
 
   unobserve() {
-    if (this.observer) {
-      this.observer.disconnect();
+    if (!this._isObservable) {
+      return;
     }
+    this.observerManager.unobserve(this.element);
   }
 
   didUpdateArguments() {
     this.unobserve();
   }
 
   didReceiveArguments() {
-    let [handler, options] = this.args.positional;
+    if (!this._isObservable) {
+      return;
+    }
+
+    let { onEnter, onExit, options } = this.args.named;
 
-    // Save arguments for when we need them
-    this.handler = handler;
-    this.options = options || this.options;
+    assert('onEnter or/and onExit is required', onEnter || onExit);
```

Comment:
I like this

Reply:
Small improvement I'd suggest (not a blocker):

```suggestion
    assert(
      'onEnter or/and onExit is required',
      typeof onEnter === 'function' ||
      typeof onExit === 'function
    );
```

That way if somebody goofs and passes `onEnter="potato"` they still get a debug assertion.